.circleci/config.yml

```diff
@@ -51,7 +55,41 @@ jobs:
       - run: yarn test --full --ci
 
 workflows:
-  version: 2
+  version: 2.0
+
+  build_without_deploy:
+    jobs:
+      - docker/publish:
+          deploy: false
+          image: sourcecred/sourcecred
+          tag: latest
+          filters:
+            branches:
+              ignore: 
+                - master
+          after_build:
+            - run:
+                name: Preview Docker Tag for Build
+                command: |
+                   DOCKER_TAG=$(docker run sourcecred/sourcecred --version | cut -d' ' -f2)
+                   echo "Version for Docker tag is ${DOCKER_TAG}"
+
+  # This workflow will deploy images on merge to master only
+  build_with_deploy:
+    jobs:
+      - docker/publish:
+          image: sourcecred/sourcecred
+          tag: latest
+          filters:
+            branches:
+             only: master
+          after_build:
+            - run:
+                name: Publish Docker Tag with Sourcecred Version
+                command: |
+                   DOCKER_TAG=$(docker run sourcecred/sourcecred --version | cut -d' ' -f2)
```

Comment:
Cool idea to use the just built container `--version` command. My inclination would be to use git tags for this though. It's pretty common for sourcecreds workflow to update master before creating releases.

Perhaps tag the master branch as `dev` and split a `v0.4.0` git tag into `latest`, `0`, `0.4` and `0.4.0`?
(The `0` one would not be too useful now, but will be after a 1.0.0 release)

Reply:
I'm not totally clear how you'd want this to work with CircleCI logic - I haven't done anything with tags [because it seems a little buggy](https://discuss.circleci.com/t/workflows-with-jobs-triggered-by-tag/18543/4) to not appear as a workflow. If you want to take a shot at editing the config, please go ahead! I wouldn't know the changes to make.

Reply:
The filter that I've used in past projects for tag only jobs:
```yml
filters:
  branches:
    ignore: /.*/
  tags:
    only: /^v.*/
```

It assumes you have a `v` prefix in the git tag for your semver releases.

Reply:
okay - I'll try this!

Reply:
The workflow I take when publishing releases is:
- create a PR which updates the version within SourceCred, e.g. https://github.com/sourcecred/sourcecred/pull/1271
- merge that PR
- create a Release via the GitHub UI

creating the release is what sets the tag, I believe on the then-master commit. However, that commit will already have been through CI. so, do we have any guarantee that the publish workflow will actually trigger?

Reply:
That's right. Branches and tags are at a high level the same thing. A pointer to a commit ID.
Pushing a tag (just like a branch) to github triggers it's own webhook to CircleCI, that's what we're listening for here, so you're guaranteed this job is run as a result of pushing the tag. You don't have to coordinate it with pushing commits to a branch.

In fact, if you ever find you've messed it up. With a bit more work you can "update" tags by deleting them upstream `git push --delete origin tagname`, deleting it locally `git tag --delete tagname`, recreating the tag and pushing it. Even if the same commit is referenced, you can force trigger the webhook again. It's made more cumbersome because conceptually tags should be immutable and you probably want a minor semver update instead.